### PR TITLE
[WIP] Migrate to metaphysics v2 schema

### PR DESCRIPTION
**Pull request by alloy**
Closes https://artsyproduct.atlassian.net/browse/PLATFORM-1446

- [x] Upgrades to Relay v5 and associated packages
- [x] Uses Relay centralised configuration
- [x] Patches Relay to print IDE friendly errors
- [x] Updates Relay artifacts
- [x] Update to latest v2 schema
- [ ] Write codemod to convert most cases automatically
- [ ] Do a `id` to `gravityID` pass, including fixtures, and get `relay-compiler` and `yarn type-check` to pass.
- [ ] Do a `id` to `internalID` pass, including fixtures, and get `relay-compiler` and `yarn type-check` to pass.
- [ ] Do a `__id` to `id` pass, including fixtures, and get `relay-compiler` and `yarn type-check` to pass.
- [ ] Get tests to pass.
- [ ] Get apps at runtime to work.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -2,36 +2,45 @@
...
-        "Components": "./src/Components",
-        "Styleguide": "./src/Styleguide",
-        "package.json": "./package.json"
+    [
+      "transform-rename-import",
+      {
+        "original": "../src/__generated__",
+        "replacement": "$1./__generated__"
+      }
+    ],
```
Comment (damassi) [inline .babelrc]: I wonder if this is still needed?

Comment (alloy) [reply .babelrc]: Probably, as this is more of a custom build setup decision we made, outside of relay's workings, right?

Comment (damassi) [reply .babelrc]: I remember a pathing hiccup prompted this, but am unsure exactly what that was. I'll try removing it in a separate branch and see what happens.

File: codemods/convert-to-metaphysics-v2.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    .find(j.TaggedTemplateExpression, {
+      tag: { type: "Identifier", name: "graphql" },
+    })
+    .replaceWith(nodePath => {
+      const { node } = nodePath
+      const { raw } = nodePath.value.quasi.quasis[0].value
+      const gqlAst = parse(raw)
+
+      // FIXME: The id case is a bit complicated and needs to be handled smarter
+      let editedAst = renameField(gqlAst, "id", "unknownIDType")
```
Comment (zephraph) [inline codemods/convert-to-metaphysics-v2.js]: We can't really do a naive update of the `id` field. We should probably take [this mapping](https://artsy.slack.com/files/U029ZQKU2/FKB3E2W49/2019-06-12_gravity_id_mappings_json.js) created by @joeyAghion and use it to selectively rename the ids to either `internalID` or `slug` respectively.

**Comment by mzikherman** [comment]
This is done'd!

**Pull request closed without merging**